content/rdi/quickstart/write-behind-guide.md

```diff
@@ -229,6 +236,106 @@ To use the metrics you can either:
+  # - table with columns
+  # - sql
+
+  #table: EMP
+  #columns:
+  # - first_name
+  # - last_name
+  # - birth_date
+
+  # bind variables are marked with a colon and must be defined in the `keys` property.
+  sql: |
+    SELECT emp.*, address.*, kids.*
+    FROM emp
+    LEFT JOIN address ON emp.employee_id = address.employee_id
+    LEFT JOIN kids ON emp.employee_id = kids.employee_id
+    WHERE emp.employee_fname= :first_name AND emp.employee_lname = :last_name;
+
+transform:
+  - uses: rename_field
+    with:
+      fields:
+        - from_field: employee_fname
+          to_field: first_name
+        - from_field: employee_lname
+          to_field: last_name
+output:
+  expire: 100
+```
+
+#### The keyspace section
+
+This section describes the Redis keyspace pattern and job type:
+
+- The `pattern` attribute that would trigger the read-through job.
+- The `trigger` attribute: indicates it is a read-through and not a write-behind job, using the `read-through` value.
+
+#### The keys section
+
+This section has the following attributes:
+
+- The `delimiter` attribute: specifies a delimiter for tokenizing the Redis key. These tokens are used with the `fields` attribute (see below).
+- Alternatively, specify the `expression` attribute. In this case, the regular expression will be used to tokenize the the Redis key. for example, `expression: "(.+):(\\d+)"` will tokenize the key to anything before the colon and a number after the colon.
+- The `fields` attribute is a map of field name and its offset in the array of tokens generated by the `expression` or or the `delimiter` attributes.
+
+#### downstream database information
+
+The `source` section must specify the following attributes:
+
+- `connection` - a connection alias, referring to a connection section in the `config.yaml` file.
+- `db` - the name of the database endpoint to connect use.
+- `schema` - the name of the schema to use.
+- `table` or `sql` attributes:
+  - `table` - the table to query.
+  - `sql` - a SQL statement in case you would like to `JOIN` several tables or to perform a different complex query. see more below about using this option.
+
+The `source` section can optionally include the following attributes:
+
+- `retries` - Number of attempts to read from the database. The default is 1.
+- `timeout` - Number of seconds to wait before reading of data is timed-out (0 - no limit). The default is 3 seconds.
+
+##### Using custom sql queries
+
+You can use any valid `SELECT` statement that can be parsed by the specific downstream database. The values to use in `WHERE` or `ON` sections can be calculated at runtime if you specify one of the keys in the `keys` section.
+
+#### Using transformations
+
+RDI supports transformation blocks with read-through jobs. You can use transformations to remove fields, map fields, add fields etc.
+Keep in mind that removing all fields or applying a filter will result in an error as no record can be written to Redis and returned to the client.
```

Comment:
consider this:

RDI provides support for transformation blocks within read-through jobs. These transformations can be utilized to perform various operations such as removing fields, mapping fields, adding fields, and more.

However, it is important to note that certain transformations, such as removing all fields or applying a filter that results in no records, will lead to an error. This occurs because there will be no valid record to write to Redis and return to the client. Therefore, it's crucial to ensure that transformations do not unintentionally result in empty or filtered data sets.

Reply:
done